src/main/java/com/commercetools/sync/cartdiscounts/CartDiscountSync.java

```diff
@@ -202,12 +200,12 @@ private void handleError(@Nonnull final String errorMessage,
                         .resolveReferences(newCartDiscount)
                         .thenCompose(resolvedDraft -> syncDraft(oldCartDiscountMap, resolvedDraft))
                         .exceptionally(completionException -> {
-                            final ReferenceResolutionException referenceResolutionException =
-                                (ReferenceResolutionException) completionException.getCause();
-                            final String errorMessage = format(FAILED_TO_RESOLVE_REFERENCES, newCartDiscount.getKey(),
-                                             referenceResolutionException.getMessage());
-                            handleError(errorMessage, referenceResolutionException, 1);
+                            final String errorMessage = format(FAILED_TO_RESOLVE_REFERENCES,
```

Comment:
`syncDraft ` might throw an exception which is not related to ReferenceResolution? It means that this message would be wrong/confusing.

Reply:
Since our approach now is to pass the raw exception to handleError(..) no matter which type it is, the program would not distinguish exception type inside exceptionally(..). How about re-defining the error message content not to be resolve references specific?